File: hw/basurto/u3/hw22MazeSolidPrinciples/Maze/Maze/src/main/java/ec/edu/espe/maze/model/Maze.java
package ec.edu.espe.maze.model;

public class Maze {
    private final int width;
    private final int height;
    private final Cell[][] grid;
    private Cell entrance;
    private Cell exit;

    public Maze(int width, int height) {
        if (width <= 0 || height <= 0) {
            throw new IllegalArgumentException();
        }
        this.width = width;
        this.height = height;
        this.grid = new Cell[height][width];
        for (int y = 0; y < height; y++) {
            for (int x = 0; x < width; x++) {
                grid[y][x] = new Cell(x, y);
            }
        }
    }

    public Cell getCell(int x, int y) {
        if (x < 0 || x >= width || y < 0 || y >= height) {
            return null;
        }
        return grid[y][x];
    }

    public void configureEntranceExit(Cell entrance, Cell exit) {
        this.entrance = entrance;
        this.exit = exit;
    }

    public Cell getEntrance() {
        return entrance;
    }

    public Cell getExit() {
        return exit;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public Cell[][] getGrid() {
        return grid;
    }
}

File: hw/artieda/u3/hw22MazeSolidPrinciplesCode/MazeJavaScript/model/maze.js
class Maze {
    constructor(width, height) {
        if (width <= 0 || height <= 0) {
            throw new Error();
        }
        this.width = width;
        this.height = height;
        this.grid = [];
        for (let y = 0; y < height; y++) {
            this.grid[y] = [];
            for (let x = 0; x < width; x++) {
                this.grid[y][x] = new Cell(x, y);
            }
        }
        this.entrance = null;
        this.exit = null;
    }

    getCell(x, y) {
        if (x < 0 || x >= this.width || y < 0 || y >= this.height) {
            return null;
        }
        return this.grid[y][x];
    }

    configureEntranceExit(entrance, exit) {
        this.entrance = entrance;
        this.exit = exit;
    }

    getEntrance() {
        return this.entrance;
    }

    getExit() {
        return this.exit;
    }

    getWidth() {
        return this.width;
    }

    getHeight() {
        return this.height;
    }

    getGrid() {
        return this.grid;
    }
}
